google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/SchemaAwareStreamWriter.java

```diff
@@ -43,6 +43,13 @@
  * StreamWriter functions, but also provides an additional feature: schema update support, where if
  * the BigQuery table schema is updated, users will be able to ingest data on the new schema after
  * some time (in order of minutes).
+ *
+ * <p>NOTE: The schema update ability will be disabled when you pass in a table schema explicitly
+ * through the writer. It is recommended that user either use JsonStreamWriter (which fully manages
+ * table schema) or StreamWriter (which accepts proto format in raw and user will handle the schema
+ * update event themsevles). If you sse this class, you need to be very cautious about possible
+ * mistmach between the writer's schema and the input data, any mismatch of the two will cause data
+ * corruption.
```

Comment:
this makes it such a bad thing to expose to the user.. are we sure any user is ever using it?

Reply:
Yeah, if your change of interface is out and there is no complain about it, we can at least change it to be non public.